modules/apps/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/util/MBMessageUtil.java

```diff
@@ -17,19 +17,175 @@
 import com.liferay.message.boards.model.MBMessage;
 import com.liferay.message.boards.service.persistence.MBMessageFinder;
 import com.liferay.message.boards.service.persistence.MBMessagePersistence;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.petra.string.StringPool;
 import com.liferay.portal.kernel.dao.orm.QueryDefinition;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.parsers.bbcode.BBCodeTranslatorUtil;
+import com.liferay.portal.kernel.service.ServiceContext;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
 import com.liferay.portal.kernel.util.ListUtil;
 import com.liferay.portal.kernel.util.OrderByComparator;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.util.WebKeys;
 import com.liferay.portal.kernel.workflow.WorkflowConstants;
 
 import java.util.Comparator;
 import java.util.List;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.apache.commons.lang.StringUtils;
 
 /**
  * @author Preston Crary
  */
 public class MBMessageUtil {
 
+	public static String getMessageBody(
+		boolean htmlFormat, MBMessage mbMessage, String quoteMark,
+		ServiceContext serviceContext) {
+
+		if (!htmlFormat) {
+			return _quotedMessage(true, mbMessage.getBody(), quoteMark);
+		}
+
+		if (!mbMessage.isFormatBBCode()) {
+			return mbMessage.getBody();
+		}
+
+		try {
+			String messageBody = BBCodeTranslatorUtil.getHTML(
+				mbMessage.getBody());
+
+			HttpServletRequest httpServletRequest = serviceContext.getRequest();
+
+			if (httpServletRequest == null) {
+				return messageBody;
+			}
+
+			ThemeDisplay themeDisplay =
+				(ThemeDisplay)httpServletRequest.getAttribute(
+					WebKeys.THEME_DISPLAY);
+
+			return MBUtil.replaceMessageBodyPaths(themeDisplay, messageBody);
+		}
+		catch (Exception exception) {
+			_log.error(
+				StringBundler.concat(
+					"Unable to parse mbMessage ", mbMessage.getMessageId(),
+					": ", exception.getMessage()));
+		}
+
+		return mbMessage.getBody();
+	}
+
+	public static String getQuoteMark(boolean htmlFormat) {
+		if (htmlFormat) {
+			return StringPool.BLANK;
+		}
+
+		return _QUOTE_MARK + StringPool.SPACE;
+	}
```

Comment:
This depends only on `htmlFormat`, so why force callers to pass this value in addition to htmlFormat? Make this method private, and as for the quote type internally.

Reply:
This applies to any method that takes both htmlFormat and quote as arguments.

Reply:
there are cases where you can not infer the value of the quote mark from the htmlFormat
so in some cases are still there both params